src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,190 @@
+        /// </remarks>
+        private void Initialize()
+        {
+            Span<byte> configurationSetting0 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR0,
+                (byte)Register.ONESHOT_FAULT_SETTING
+            };
+            Span<byte> configurationSetting1 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR1,
+                _thermocoupleType
+            };
+
+            Write(configurationSetting0);
+            Write(configurationSetting1);
+
+        }
+
+        /// <summary>
+        /// Command to Get Temperature from the Device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data for the cold junction temperature also checks for errors to throw
+        /// </remarks>
+        private double ReadCJTemperature()
+        {
+            var spiOutputData = WriteRead(Register.READ_CR0, 16);
+            var coldJunctionTemperature = ConvertspiOutputDataTempColdJunction(spiOutputData);
+            return coldJunctionTemperature;
+        }
+
+        /// <summary>
+        /// Converts the Thermocouple Temperature Reading
+        /// </summary>
+        /// <remarks>
+        /// Takes the spi data as an input and outputs the Thermocouple Temperature Reading
+        /// </remarks>
+        private Temperature ReadTemperature(out byte[] spiOutputData)
+        {
+            spiOutputData = WriteRead(Register.READ_CR0, 16);
+            double tempRaw = (((spiOutputData[13] & 0x7F) << 16) + (spiOutputData[14] << 8) + (spiOutputData[15]));
+            double temperature = tempRaw / 4096; // temperature in degrees C
+            if ((spiOutputData[13] & 0x80) == 0x80) // checks if the temp is negative
+            {
+                temperature = -temperature;
+            }
+
+            var temperatureOut = Temperature.FromDegreesCelsius(temperature); // Converts temperature to type Temperature struct and establishes it as Degrees C for its initial units
+
+            return temperatureOut;
+        }
+
+        /// <summary>
+        /// Converts Cold Junction Temperature Reading
+        /// </summary>
+        /// <remarks>
+        /// Takes the spi data as an input and outputs the Cold Junction Temperature Reading
+        /// </remarks>
+        /// <param name="spiOutputData">Spidata read from the device as 16 bytes</param>
+        private double ConvertspiOutputDataTempColdJunction(byte[] spiOutputData)
+        {
+            var tempRaw = (((spiOutputData[11] & 0x7F) << 8) + spiOutputData[12]);
+            double temperatureColdJunction = tempRaw / 256; // convert decimal ouput to temperature using a constant from the Technical Manual
+            if ((spiOutputData[11] & 0x80) == 0x80) // checks if the temp is negative
+            {
+                temperatureColdJunction = -temperatureColdJunction;
+            }
```

Comment:
This should be something along the lines of:

```csharp
Span<byte> reading = new Span<byte>(spiOutputData, 11, 2);
short tempRaw = BinaryPrimitives.ReadInt16BigEndian(reading);
return tempRaw / 256.0;
```

note: `256.0` not `256`, dividing integer by integer will lose everything after decimal separator

Reply:
I updated the code. The only issue is that the previous code removes the first two digits of byte 11 so I have to code that in to read the proper temperature. 

reading[0] = (byte)(spiOutputData[11] & 0x3F);

Reply:
Looking at the documentation I have some questions (https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf page 24):
- why 11 and 12, the doc says about 0xA and 0xB which maps to 10 and 11
- the way spec describes this the code should be more like (written by hand, not tested):

```csharp
byte msb = ...; // 0xA
byte lsb = ...; // 0xB

bool sign = (msb >> 7) == 1;
int raw = ((msb & 0x7F) << 6) | (lsb >> 2);
if (sign)
{
    raw = -raw;
}

return raw / 64.0;
```

Reply:
the var spiOutputData = WriteRead(Register.READ_CR0, 16); reads 16 bits starting at 1. So it shifts the reading by 1